src/calendar-api.ts

```diff
@@ -0,0 +1,56 @@
+//function to fetch events from cloud API endpoint
+
+export const fetchGoogleCalendarEvents = async () => {
+	const apiKey = import.meta.env.VITE_AWS_API_KEY; //secured API key
+	const apiUrl = "https://ghvy6isiq6.execute-api.eu-north-1.amazonaws.com/dev/events"; //AWS API url
+
+	interface CalendarEvent {
+		start?: {
+			dateTime?: string; // ISO for events with specific times
+			date?: string; // ISO for all-day events
+		};
+		end?: {
+			dateTime?: string; // ISO for events with specific times
+			date?: string; // ISO for all-day events
+		};
+	}
```

Comment:
Type definitions should be outside of functions, or in separate files all together. Also unless `interface` is absolutely necessary in some Object Orientated Programming case you should use `type`. You can read why on the internet/ChatGPT or we can talk about it when we see each other next time.

Reply:
Ok, corrected.